test_fms/mpp/test_stdin.F90

```diff
@@ -0,0 +1,47 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @file
+!! @brief Unit test for the STDIN function
+!! @author Colin Gladue
+!! @email gfdl.climate.model.info@noaa.gov
+
+program test_stdin
+  use mpp_mod, only : mpp_error, FATAL
+  use mpp_mod, only : stdin
+  use iso_fortran_env, only : INPUT_UNIT
+
+  integer :: in_unit !< Stores the returned standard input unit number
+
+  in_unit = stdin()
+
+  write(*,*) "stdin() should get and return the value of in_unit."
```

Comment:
Instead of `write(*,*)` you can also use `OUTPUT_UNIT` from iso_fortran_env, and then do `write(OUTPUT_UNIT,*)` to print to stdout.

Reply:
These changes have been made, but this change request still seems to be blocking the merge.